linode/linode-cli
# POC - Add BATS tests

* This PR adds some initial support for BATS (bash automated testing system) test files.
* Added bats-assert and bats-support sub-modules
* Added some initial test cases for the /linodes endpoint
* Added a `common.bash` file that will contain common helper functions used in other bats test files.

# Testing

1. Install BATS (on mac, `brew install bats-core` or see : https://github.com/bats-core/bats-core/#installation) for non-mac installs
2. checkout this pr
3. `git submodule init && git submodule update`
4. Run the example test from the base of the repo: `bats test/linodes`
⚠️ The test suite will delete all Linodes on the account that runs the tests, even those that existed before the tests were run ⚠️ 
5. Observe, _*tests*_!

# To Do

* Document how to install bats and run the tests
* Update tests to not assume a `.linode-cli` file exists
* Make root password random, not hard coded
